apps/chat/conversation.py

```diff
@@ -193,10 +195,15 @@ def _get_new_summary(llm, pruned_memory, summary, max_token_limit):
     recursively call this function with the remaining memory."""
     tokens, context = _get_summary_tokens_with_context(llm, summary, pruned_memory)
     next_batch = []
-    while tokens > max_token_limit:
+    while pruned_memory and tokens > max_token_limit:
         next_batch.insert(0, pruned_memory.pop())
         tokens, context = _get_summary_tokens_with_context(llm, summary, pruned_memory)
 
+    if not context["new_lines"]:
```

Comment:
Can we assume that `pruned_memory` will always be populated if we call this function? If not, then the possibility exists where if `pruned_memory` is empty, then the call on line 196 will return a context where `new_lines` will be empty and we will skip the while-loop, so the check on 202 will evaluate to True and the `summary-too-large` error will be logged. I'm just wondering if this will always be accurate?